### PR TITLE
Chef 17 Compatability

### PR DESCRIPTION
**Pull request by larry-herz**
# Description

Updates to chef 17 compatibility

## Issues Resolved

Removes .delivery dir tree and adds reusable tests.  Adds unified_mode true flag. Updates kitchen files with supported OS versions. 

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: kitchen.yml [hunk 1]
```diff
@@ -9,11 +9,6 @@ provisioner:
   name: chef_zero
 
 platforms:
-  - name: windows-2012r2
```
Comment (tas50) [inline kitchen.yml:12]: Sadly 2012r2 is still a thing. https://docs.microsoft.com/en-us/lifecycle/products/windows-server-2012-r2

File: spec/unit/recipes/default_spec.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
 describe 'winrm::default' do
   context 'Windows' do
     let(:chef_run) do
-      runner = ChefSpec::ServerRunner.new(file_cache_path: Chef::Config[:file_cache_path], platform: 'windows', version: '2012R2')
+      runner = ChefSpec::ServerRunner.new(file_cache_path: Chef::Config[:file_cache_path], platform: 'windows', version: '2019')
```
[the remark this answers is not in the document]
Comment (larry-herz) [reply spec/unit/recipes/default_spec.rb]: Removed version reference.

File: kitchen.yml (hunk 1 above)
Comment (larry-herz) [reply kitchen.yml]: Reverted changes that dropped 2012.

File: kitchen.appveyor.yml [hunk 3]
```diff
@@ -19,9 +19,9 @@ provisioner:
   chef_license: accept
```
[the remark this answers is not in the document]
Comment (larry-herz) [reply kitchen.appveyor.yml]: Not sure who needs to make this call.

[the remark this answers is not in the document]
Comment (damacus) [reply kitchen.appveyor.yml]: Yeah we can remove this. When we readd window testing we should do it via GitHub actions

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -8,34 +8,11 @@ name: ci
...
-
-  mdl:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check out code
-        uses: actions/checkout@v2
-      - name: Run Markdown Lint
-        uses: actionshub/markdownlint@main
+  lint-unit:
+    uses: sous-chefs/.github/.github/workflows/lint-unit.yml@0.0.6
```
Comment (ramereth) [inline .github/workflows/ci.yml:12]: Please use this version until we get the other issue fixed.
```suggestion
    uses: sous-chefs/.github/.github/workflows/lint-unit.yml@0.0.3
```

**Review by ramereth — CHANGES_REQUESTED** [review]
Please remove all appveyor and azure kitchen references and use Github Actions in a similar fashion we do here: https://github.com/sous-chefs/ark/blob/main/.github/workflows/ci.yml#L49-L72

**Review by damacus — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [4.0.0](https://github.com/sous-chefs/winrm/releases/tag/4.0.0)

**Pull request merged**